File: prompts/fixtures/js/debug_crash.js
"use strict";

function divide(a, b) {
    if (b === 0) throw new Error("Division by zero");  // line 4
    return Math.floor(a / b);                           // line 5
}

function dereferenceNull() {
    const obj = null;              // line 9
    console.log(obj.value);        // line 10 — TypeError
}

function abortHandler() {
    process.abort();               // line 14 — SIGABRT
}

function main() {
    if (process.argv.length < 3) {                              // line 18
        console.log(`Usage: node ${process.argv[1]} [divzero|null|abort]`);  // line 19
        process.exit(1);                                        // line 20
    }
    const mode = process.argv[2];                              // line 22
    console.log(`mode: ${mode}`);                              // line 23
    if (mode.startsWith("d")) {                                // line 24
        divide(10, 0);                                         // line 25
    } else if (mode.startsWith("n")) {                         // line 26
        dereferenceNull();                                     // line 27
    } else if (mode.startsWith("a")) {                         // line 28
        abortHandler();                                        // line 29
    }
}

main();

File: prompts/fixtures/java/DebugCrash.java
public class DebugCrash {
    static int divide(int a, int b) {
        return a / b;                                 // line 3 — ArithmeticException when b=0
    }

    static void dereferenceNull() {
        String s = null;                              // line 7
        System.out.println(s.length());               // line 8 — NullPointerException
    }

    static void abortHandler() {
        throw new RuntimeException("abort requested");// line 12
    }

    public static void main(String[] args) {
        if (args.length < 1) {                        // line 16
            System.out.printf("Usage: DebugCrash [divzero|null|abort]%n"); // line 17
            System.exit(1);                           // line 18
        }
        System.out.printf("mode: %s%n", args[0]);    // line 20
        if (args[0].startsWith("d")) {                // line 21
            divide(10, 0);                            // line 22
        } else if (args[0].startsWith("n")) {         // line 23
            dereferenceNull();                        // line 24
        } else if (args[0].startsWith("a")) {         // line 25
            abortHandler();                           // line 26
        }
    }
}
